core/src/main/java/io/questdb/griffin/engine/ops/CreateTableOperationImpl.java

```diff
@@ -534,6 +535,10 @@ public void validateAndUpdateMetadataFromSelect(RecordMetadata metadata) throws
             final String columnName = metadata.getColumnName(i);
             final TableColumnMetadata augMeta = augmentedColumnMetadata.get(columnName);
 
+            if (!TableUtils.isValidColumnName(columnName, Integer.MAX_VALUE)) {
```

Comment:
Max length `Integer.MAX_VALUE` is way too much, in most of the file systems it's 255. Either change to 255 here or pass the FS specific from the configuration.